File: Java/src/JSJ/Math/BaseMath.java
package JSJ.Math;

class BaseMath{

    /**
     * Adds two numbers.
     *
     * @param a First number.
     * @param b Second number.
     * @return Sum of a and b.
     */
    public static double add(double a, double b) {
        return a + b;
    }

    /**
     * Subtracts two numbers.
     *
     * @param a First number.
     * @param b Second number.
     * @return Difference of a and b.
     */
    public static double subtract(double a, double b) {
        return a - b;
    }

    /**
     * Multiplies two numbers.
     *
     * @param a First number.
     * @param b Second number.
     * @return Product of a and b.
     */
    public static double multiply(double a, double b) {
        return a * b;
    }

    /**
     * Divides two numbers.
     *
     * @param a Dividend.
     * @param b Divisor.
     * @return Quotient of a divided by b.
     * @throws ArithmeticException If b is zero.
     */
    public static double divide(double a, double b) {
        if (b == 0) {
            throw new ArithmeticException("Division by zero is not allowed.");
        }
        return a / b;
    }

    /**
     * Calculates the power of a number.
     *
     * @param base The base number.
     * @param exponent The exponent.
     * @return Result of base raised to the power of exponent.
     */
    public static double power(double base, double exponent) {
        return Math.pow(base, exponent);
    }

    /**
     * Calculates the square root of a number.
     *
     * @param a The number.
     * @return Square root of a.
     */
    public static double squareRoot(double a) {
        if (a < 0) {
            throw new ArithmeticException("Square root of negative numbers is not supported.");
        }
        return Math.sqrt(a);
    }

    /**
     * Calculates the factorial of a number.
     *
     * @param n Non-negative integer.
     * @return Factorial of n.
     * @throws IllegalArgumentException If n is negative.
     */
    public static long factorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("Factorial of negative numbers is not defined.");
        }
        long result = 1;
        for (int i = 2; i <= n; i++) {
            result *= i;
        }
        return result;
    }

    /**
     * Calculates the greatest common divisor (GCD) of two integers.
     *
     * @param a First integer.
     * @param b Second integer.
     * @return GCD of a and b.
     */
    public static int gcd(int a, int b) {
        if (b == 0) {
            return Math.abs(a);
        }
        return gcd(b, a % b);
    }
}

File: TypeScript/JSTS/Math/baseMath.ts
// File: MathUtils.ts

export class MathUtils {
    // Adds two numbers
    static add(a: number, b: number): number {
        return a + b;
    }

    // Subtracts the second number from the first
    static subtract(a: number, b: number): number {
        return a - b;
    }

    // Multiplies two numbers
    static multiply(a: number, b: number): number {
        return a * b;
    }

    // Divides the first number by the second
    // Throws an error if division by zero is attempted
    static divide(a: number, b: number): number {
        if (b === 0) {
            throw new Error("Division by zero is not allowed.");
        }
        return a / b;
    }

    // Finds the modulus (remainder) of the division of two numbers
    static modulus(a: number, b: number): number {
        return a % b;
    }

    // Calculates the power of a number (a^b)
    static power(base: number, exponent: number): number {
        return Math.pow(base, exponent);
    }

    // Returns the absolute value of a number
    static absolute(value: number): number {
        return Math.abs(value);
    }

    // Finds the square root of a number
    // Throws an error if the input is negative
    static squareRoot(value: number): number {
        if (value < 0) {
            throw new Error("Square root of negative numbers is not supported.");
        }
        return Math.sqrt(value);
    }

    // Rounds a number to the nearest integer
    static round(value: number): number {
        return Math.round(value);
    }

    // Generates a random number between two values (inclusive)
    static randomBetween(min: number, max: number): number {
        return Math.random() * (max - min) + min;
    }
}
